### PR TITLE
Only update when json_metadata changes

### PR DESCRIPTION
**Pull request by bbaker6225**
Also removes old logging

## Description

Description of changes made

## Ticket Link

Link to the associated ticket
[Link title](https://openedx.atlassian.net/browse/ENT-XXXX)

## Post-review

Squash commits into discrete sets of changes

### CODE REVIEW

File: enterprise_catalog/apps/catalog/models.py [hunk 1]
```diff
@@ -287,10 +300,17 @@ def associate_content_metadata_with_query(metadata, catalog_query):
 
     Returns set of content_keys
     """
-    number_of_items_that_have_not_changed = 0
     content_keys = set()
     for entry in metadata:
         content_key = get_content_key(entry)
+        if content_key in content_keys:
```
Comment (bbaker6225) [inline enterprise_catalog/apps/catalog/models.py:306]: Moved this to the top so we can just `continue` for metadata that hasn't changed. Every entry should have its content key added so doesn't change any logic

File: enterprise_catalog/apps/catalog/models.py [hunk 2]
```diff
@@ -303,8 +323,9 @@ def associate_content_metadata_with_query(metadata, catalog_query):
             old_metadata = None
 
         if old_metadata:
-            if sorted(json.dumps(entry)) == sorted(json.dumps(old_metadata.json_metadata)):
-                number_of_items_that_have_not_changed += 1
+            if get_sorted_string_from_json(entry) == get_sorted_string_from_json(old_metadata.json_metadata):
+                # Only update the existing ContentMetadata object if its json has not changed
```
Comment (adamstankiewicz) [inline enterprise_catalog/apps/catalog/models.py]: nit: should this comment say "if its json **has** changed"?

**Review by adamstankiewicz — APPROVED** [review]
Just one nit about a comment, but otherwise LGTM.

File: enterprise_catalog/apps/catalog/models.py (hunk 2 above)
Comment (bbaker6225) [reply enterprise_catalog/apps/catalog/models.py]: Oops, yes good catch

**Pull request merged**